querier.go

```diff
@@ -60,6 +63,28 @@ func (q *querier) LabelValues(n string) ([]string, error) {
 	return q.lvals(q.blocks, n)
 }
 
+// LabelNames returns all the unique label names present querier blocks.
+func (q *querier) LabelNames() ([]string, error) {
+	labelNamesMap := make(map[string]struct{})
+	for _, b := range q.blocks {
+		names, err := b.LabelNames()
+		if err != nil {
+			return nil, errors.Wrap(err, "LabelNames() from IndexReader")
```

Comment:
Well technically it is from Querier interface. The fact that it Is  from index reader is just assuming implementation. Worth to maybe change this to Querier?

Reply:
That is a typo (I moved the previous implementation and forgot to change the message). Thanks!